pkg/destroy/baremetal/baremetal.go

```diff
@@ -36,8 +43,66 @@ func (o *ClusterUninstaller) Run() error {
 // New returns bare metal Uninstaller from ClusterMetadata.
 func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.Destroyer, error) {
 	return &ClusterUninstaller{
+		ClusterID:               metadata.ClusterID,
 		LibvirtURI:              metadata.ClusterPlatformMetadata.BareMetal.LibvirtURI,
 		BootstrapProvisioningIP: metadata.ClusterPlatformMetadata.BareMetal.BootstrapProvisioningIP,
 		Logger:                  logger,
 	}, nil
 }
+
+// deleteStoragePool destroys, deletes and undefines any storagePool left behind during the creation
+// of the bootstrap VM
+func (o *ClusterUninstaller) deleteStoragePool(conn *libvirt.Connect) error {
+	o.Logger.Debug("Deleting baremetal bootstrap volumes")
+
+	pools, err := conn.ListStoragePools()
+	if err != nil {
+		return errors.Wrap(err, "list storage pools")
+	}
+
+	for _, pname := range pools {
+		// delete only pool names that contains cluster name and 'bootstrap' string
+		if !strings.Contains(pname, "-bootstrap") && !strings.Contains(pname, o.ClusterID) {
```

Comment:
The pool name would be verbatim `ostest-bootstrap`, right? Can we do an exact equality match?

```go
if pname == fmt.Sprintf("%s-bootstrap", o.ClusterID) {
    // ....
}
```

If I create two test clusters, `ostest-bootstrap` and `ostest2-boostrap`, your check on ostest-bootstrap would match both, and we'd have a very unhappy developer.

Reply:
The only reason I kept it this way was due to the fact that any pools not automatically cleaned up will just linger around forever. Someone has to clean up later using `virsh pool-delete`. In any case, I can see this causing problems if a current installation is progressing when invoking the destroy...